File: draw.rb
require 'java'

java_import 'javax.swing.JFrame'
java_import 'javax.swing.JPanel'
java_import 'java.awt.Dimension'
java_import 'java.awt.Graphics'
java_import 'java.awt.event.MouseEvent'
java_import 'java.awt.event.MouseMotionListener'


class DrawPanel < JPanel
  
  attr_accessor :x, :y
  
  def initialize
    @x, @y = 0, 0
  end
  
  def paintComponent(graphics)
    graphics.draw_oval(@x, @y, 10, 10)
  end
  
end


class MyListener
  
  attr_accessor :panel
  
  def mouseDragged(event)
    @panel.x = event.x
    @panel.y = event.y
    @panel.repaint
  end
  
  def mouseMoved(event)
  end
  
end


panel = DrawPanel.new
listener = MyListener.new
listener.panel = panel
panel.add_mouse_motion_listener(listener)

frame = JFrame.new 'JRuby Paint'
frame.size = Dimension.new 640, 480
frame.add panel
frame.visible = true

File: Draw.java
import javax.swing.JFrame;
import javax.swing.JPanel;
import java.awt.Dimension;
import java.awt.Graphics;
import java.awt.event.MouseEvent;
import java.awt.event.MouseMotionListener;

public class Draw {
    public static void main(String[] args) {
        JFrame frame = new JFrame("Java Paint");
        frame.setSize(new Dimension(640, 480));
        DrawPanel panel = new DrawPanel();
        MyListener listener = new MyListener();
        listener.setPanel(panel);
        panel.addMouseMotionListener(listener);
        frame.add(panel);
        frame.setVisible(true);
    }
}

class DrawPanel extends JPanel {
    private int x = 0;
    private int y = 0;
    public void setX(int x) {this.x = x;}
    public void setY(int y) {this.y = y;}  
    public void paintComponent(Graphics graphics) {
        graphics.drawOval(x, y, 10, 10);
    }
}

class MyListener implements MouseMotionListener {
    DrawPanel panel;
    public void setPanel(DrawPanel panel) {this.panel = panel;}
    public void mouseDragged(MouseEvent event) {
        panel.setX(event.getX());
        panel.setY(event.getY());
        panel.repaint();
    }
    public void mouseMoved(MouseEvent event) {}
}
